File: thousands-services/IvsIdleGameShared/Models/Sponsorship/SponsoredEvent.cs
using System.Text.Json.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace IvsIdleGameShared.Models.Sponsorship;

[BsonIgnoreExtraElements]
public class SponsoredEvent
{
    [BsonId]
    [BsonElement("_id")]
    [JsonIgnore]
    public ObjectId Id { get; set; }

    [JsonPropertyName("_id")]
    public string IdString => Id.ToString();

    [BsonElement("sponsorshipSlots")]
    public List<SponsorshipSlot> SponsorshipSlots { get; set; } = new List<SponsorshipSlot>();
}

[BsonIgnoreExtraElements]
public class SponsorshipSlot
{
    [BsonElement("maxSlots")]
    public int MaxSlots { get; set; } = 0;

    [BsonElement("creditsPrice")]
    public int CreditsPrice { get; set; } = 0;

    [BsonElement("usdcPrice")]
    public int UsdcPrice { get; set; } = 0;

    [BsonElement("packageDescription")]
    public string PackageDescription { get; set; } = string.Empty;

    [BsonElement("baseWC")]
    public int BaseWC { get; set; } = 0;

    [BsonElement("tier")]
    public int Tier { get; set; } = 0;

    [BsonElement("house")]
    public int House { get; set; } = 0;

    [BsonElement("sponsorshipSlotId")]
    public ObjectId SponsorshipSlotId { get; set; }
}

File: thousands-tv/wildcard/packages/interfaces/src/db/interfaces/sponsoredEvent.ts
import { Types } from "mongoose";

export const SPONSORED_EVENTS_TABLE_NAME = "sponsored-events";

export interface ISponsorshipSlot {
    maxSlots: number;
    creditsPrice: number;
    usdcPrice: number;
    packageDescription: string;
    baseWC: number;
    tier: number;
    house: number;
    sponsorshipSlotId: string;
}

export interface ISponsoredEvent {
    name: string;
    type: string;
    startTime: Date;
    sponsorLockTime: Date;
    sponsorshipSlots: ISponsorshipSlot[];
    totalWC: number;
    _id?: Types.ObjectId;
    __v?: number;
    createdAt?: Date;
    updatedAt?: Date;
}
